### PR TITLE
docs: all HTTP APIs

### LINKED ISSUE #1097: Reference for all HTTP APIs
* `/sql`
* `/promql`
* influxdb line protocol
* ....

### PR DESCRIPTION
**Pull request by nicecui**
## What's Changed in this PR

- merge the HTTP API document from `/reference` into the HTTP API document in the protocol section.
- add all user APIs
- close #1097 

Preview URL: https://refine-http-api.greptime-docs.pages.dev/nightly/user-guide/protocols/http

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]

File: docs/user-guide/protocols/http.md [hunk 1]
```diff
@@ -27,69 +41,149 @@ In this example, `Z3JlcHRpbWVfdXNlcjpncmVwdGltZV9wd2Q=` represents the Base64-en
 InfluxDB uses its own authentication format, see [InfluxDB](./influxdb-line-protocol.md) for details.
 :::
 
-### Time zone
+### Timeout
 
-GreptimeDB supports the `X-Greptime-Timezone` header in HTTP requests.
-It is used to specify the timezone for the current SQL query.
+GreptimeDB supports the `X-Greptime-Timeout` header in HTTP requests.
+It is used to specify the timeout for the request.
```
Comment (killme2008) [inline docs/user-guide/protocols/http.md:47]:
```suggestion
It is used to specify the timeout for the request running in the database server.
```

File: docs/user-guide/protocols/http.md [hunk 2]
```diff
@@ -27,69 +41,149 @@ In this example, `Z3JlcHRpbWVfdXNlcjpncmVwdGltZV9wd2Q=` represents the Base64-en
...
-For example, to insert data into the `monitor` table, use the following command:
+To submit a SQL query to the GreptimeDB server via HTTP API, use the following format:
+
+```shell
+curl -X POST \
+  -H 'Authorization: Basic {{authentication}}' \
+  -H 'X-Greptime-Timeout: {{time precision}}' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'sql={{SQL-statement}}' \
+http://{{API-host}}/v1/sql?db={{db-name}}
```
Comment (killme2008) [inline docs/user-guide/protocols/http.md:142]: The `db` is optional, so i think we don't need to present it in the sample:

```suggestion
http://{{API-host}}/v1/sql
```

File: docs/user-guide/query-data/promql.md [hunk 3]
```diff
@@ -26,6 +26,60 @@ of these API.
...
+    --data-urlencode 'query=irate(process_cpu_seconds_total[1h])' \
+    --data-urlencode 'start=2024-11-24T00:00:00Z' \
+    --data-urlencode 'end=2024-11-25T00:00:00Z' \
+    --data-urlencode 'step=1h' \
+    --data-urlencode 'db=public' \
+    http://localhost:4000/v1/prometheus/api/v1/query_range
+```
+If authentication is enabled in GreptimeDB, the authentication header is required. Refer to the [authentication documentation](/user-guide/protocols/http.md#authentication) for more details.
+
+The query string parameters for the API are identical to those of the original Prometheus API, with the exception of the additional `db` parameter, which specifies the GreptimeDB database name.
```
Comment (killme2008) [inline docs/user-guide/query-data/promql.md:43]:
```suggestion
The query string parameters for the API are identical to those of the original [Prometheus API](https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries), with the exception of the additional `db` parameter, which specifies the GreptimeDB database name.
```

**Review by killme2008 — COMMENTED** [review]
Excellent work! Really appreciate it. I have a small suggestion to add.

**Comment by nicecui** [comment]
@killme2008 PTAL

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**